File: spotlight/src/main/java/com/takusemba/spotlight/OnTargetStateChangedListener.java
package com.takusemba.spotlight;

import com.takusemba.spotlight.target.Target;

/**
 * On Target State Changed Listener
 **/
public interface OnTargetStateChangedListener<T extends Target> {
  /**
   * Called when Target is started
   */
  void onStarted(T target);

  /**
   * Called when Target is started
   */
  void onEnded(T target);
}

File: spotlight/src/main/java/com/takusemba/spotlight/OnTargetStateChangedListener.kt
package com.takusemba.spotlight

import com.takusemba.spotlight.target.Target

/**
 * On Target State Changed Listener
 */
interface OnTargetStateChangedListener<T : Target<T>> {
  /**
   * Called when Target is started
   */
  fun onStarted(target: T)

  /**
   * Called when Target is started
   */
  fun onEnded(target: T)
}
